server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/http/SessionRoutesTest.scala

```diff
@@ -0,0 +1,97 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.nio.charset.StandardCharsets
+
+import io.restassured.RestAssured
+import io.restassured.builder.RequestSpecBuilder
+import io.restassured.config.EncoderConfig.encoderConfig
+import io.restassured.config.RestAssuredConfig.newConfig
+import io.restassured.http.ContentType
+import org.apache.http.HttpStatus
+import org.apache.james.core.Username
+import org.apache.james.jmap.http.SessionRoutesTest.{BOB, TEST_CONFIGUIRATION}
+import org.apache.james.jmap.{JMAPConfiguration, JMAPServer, Serializer}
+import org.apache.james.mailbox.MailboxSession
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito._
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import org.scalatestplus.mockito.MockitoSugar
+import play.libs.Json
+import reactor.core.publisher.Mono
+
+import scala.jdk.CollectionConverters._
+
+object SessionRoutesTest {
+  private val JMAP_SESSION = "/jmap/session"
+  private val TEST_CONFIGUIRATION = JMAPConfiguration.builder.enable.randomPort.build
+  private val BOB = Username.of("bob@james.org")
+}
+
+class SessionRoutesTest extends FlatSpec with BeforeAndAfter with MockitoSugar with Matchers {
+
+  var jmapServer: JMAPServer = _
+
+  before {
+    val mockedSession = mock[MailboxSession]
+    when(mockedSession.getUser)
+      .thenReturn(BOB)
+
+    val mockedAuthFilter = mock[AuthenticationReactiveFilter]
```

Comment:
I used mock because of setting a real authentication method requires lot of works, it should be done some where in the integration tests.

Reply:
Please then consider using a lambda

Reply:
I dont get it, what lamda? a library? java 8 lamda?

Reply:
Well, in Java I woulde write:

`AuthenticationReactiveFilter authFilter = req -> MailboxSessionUtil.create(Username.of("bob"));` 

I guess there is a scala equivalent for "functional interfaces as lambdas."

Reply:
Sorry, it's class now: https://github.com/linagora/james-project/blob/master/server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/http/Authenticator.java